### PR TITLE
add form shortcuts, min vol input

### PR DESCRIPTION
**Pull request by smk762**
closes: https://github.com/KomodoPlatform/atomicDEX-Desktop/issues/1941
closes: https://github.com/KomodoPlatform/atomicDEX-Desktop/issues/1459

https://user-images.githubusercontent.com/35845239/185878663-e3e08cd1-ee8e-45ce-ba29-f59bce54d54f.mp4

### CODE REVIEW

**Review by SirSevenG — APPROVED** [review]

**Comment by tonymorony** [comment]
Thats a great feature, thank you!

1) I've noticed that inaccuracy accumulating when I press +1% (I assume it should be around 8% when I click 8 times but it's 8.32% what is quite huge diff)

https://user-images.githubusercontent.com/24797699/186668186-6d7e0a7e-6ee2-424e-84d2-6672569a2d19.mov

2) For a new user it might be not clear what -1% 0% 1% buttons means, it would be awesome to add a tooltip with clarification

**Review by tonymorony — CHANGES_REQUESTED** [review]
^^ it worth to double-check calculations, seems to be not precise

**Comment by smk762** [comment]
> ^^ it worth to double-check calculations, seems to be not precise

At the moment, the +1% is relative to currently input price, so it is compounding.
If you think its better to make it a consistent 1% relative to market, I can change that easy enough.

The only problem with this is if the coin has no cex price to calculate from. In this case, I can revert to compounding relative to price input field.

**Comment by smk762** [comment]
non-compounding is more intuitive, tho with tooltip explaining the compounding way for both priced and unpriced would be more consistent (otherwise long tooltip with conditionals).

Another alternative could be to remove those buttons for unpriced pairs (right now clicking the zero on an unpriced coin is pointless)

**Review by tonymorony — DISMISSED** [review]

**Review by Canialon — CHANGES_REQUESTED** [review]
@smk762 this tip is a good idea, but they covered the field with information. So to see how the amount changed in this field I need to move out the mouse every time. Can you please replace the tips on the bottom of the buttons? I hope it will be better

image

________

I can set a negative price if I enter 0 and click the -1 button
image
Anyway, this swap will fail because of minimum price( `[{"error":"rpc:212] dispatcher_legacy:160] lp_ordermatch:3565] lp_ordermatch:3646] Price is too low, minimum is 0.00000001"}]` ). But I guess it would be good to eliminate the possibility of setting a negative value

**Review by Canialon — DISMISSED** [review]

**Comment by tonymorony** [comment]
found a bug I think:
I can see "Min volume" I've input when I deactivate "Use custom minimum trade amount" - I assume min trade volume shouldn't apply if deactivated:

https://user-images.githubusercontent.com/24797699/187721663-447a1ce0-cf1e-47f9-8b90-44bc9425a9b3.mov

**Review by tonymorony — CHANGES_REQUESTED** [review]
would be great to fix ^^

File: atomic_defi_design/Dex/Exchange/ProView/PlaceOrderForm/OrderForm.qml [hunk 1]
```diff
@@ -108,26 +127,100 @@ ColumnLayout
...
+                let volume = API.app.trading_pg.max_volume * 0.20
+                if (volume > parseFloat(input_volume.text))
+                    volume = parseFloat(input_volume.text)
+                setMinimumAmount(General.formatDouble(volume))
+            }
+            fiat_value: General.getFiatText(non_null_volume, left_ticker)
+            left_label: "5%"
+            middle_label: "10%"
+            right_label: "20%"
+            left_tooltip_text: "Minimum accepted trade equals 5% of your balance."
```
[the remark this answers is not in the document]
Comment (smk762) [reply atomic_defi_design/Dex/Exchange/ProView/PlaceOrderForm/OrderForm.qml:212]: These buttons currently calculate the input as percentage of tradable balance - should I change them to calculate input as percentage of volume input?

**Comment by smk762** [comment]
I've also made the form stop bouncing around when you toggle the min volume field, get an error message or the "Order selected" box is displayed. Previously, when these became visible, it would shift other form elements - now everything should stay where it is when these things change between visible or hidden.

**Review by Canialon — APPROVED** [review]

**Review by tonymorony — APPROVED** [review]

**Pull request merged**